File: app/src/main/java/com/gallopdevs/athanhelper/clock/DayViewAdapter.java
package com.gallopdevs.athanhelper.clock;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.Calendar;

/**
 * Created by jgell on 5/23/2017.
 */

public class DayViewAdapter extends FragmentPagerAdapter {

    private static final String TAG = "DayViewAdapter";

    private static final int NUM_ITEMS = 7;

    public DayViewAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        Fragment fragment = new DayViewFragment();
        Bundle bundle = new Bundle();
        Calendar c = Calendar.getInstance();
        int day = c.get(Calendar.DAY_OF_WEEK);
        int indicator = ClockFragment.getNextTime();
        bundle.putInt("day", day + i);
        bundle.putInt("count", i);
        bundle.putInt("indicator", indicator);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public int getCount() {
        return NUM_ITEMS;
    }

    @Override
    public int getItemPosition(@NonNull Object object) {
        return POSITION_NONE;
    }
}

File: app/src/main/java/com/gallopdevs/athanhelper/clock/DayViewAdapter.kt
package com.gallopdevs.athanhelper.clock

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import java.util.Calendar

class DayViewAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {

    override fun createFragment(i: Int): Fragment {
        val fragment = DayViewFragment()

        val c = Calendar.getInstance()
        val weekDay = c.get(Calendar.DAY_OF_WEEK)
        val month = c.get(Calendar.MONTH) + 1
        val dayOfMonth = c.get(Calendar.DAY_OF_MONTH)

        val bundle = Bundle().apply {
            putInt(WEEK_DAY, weekDay + i)
            putInt(MONTH, month)
            putInt(DAY_OF_MONTH, dayOfMonth + i)
            putInt(PAGE_INDEX, i)
        }

        fragment.arguments = bundle
        return fragment
    }

    override fun getItemCount(): Int = NUM_ITEMS

    companion object {
        private const val NUM_ITEMS = 7
        const val PAGE_INDEX = "PAGE_INDEX"
        const val WEEK_DAY = "WEEK_DAY"
        const val DAY_OF_MONTH = "DAY_OF_MONTH"
        const val MONTH = "MONTH"
    }
}
